### DISCUSSION TITLE
Unit testing a query requires a real store

### DISCUSSION THREAD

**Discussion by JohanHeyvaert** [salesforce/akita#683] [Q&A]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[x] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior
Unit testing an akita query seems to require a real store. At least [the documentation](https://datorama.github.io/akita/docs/angular/tests/) doesn't show how to do it and in the [source code](https://github.com/datorama/akita/blob/master/libs/akita/src/lib/query.ts) I see that Query uses an internal function (_select) of the store, so probably not a good idea to mock this.

Our current tests, with a real store, are working, but I'm just wondering if this is a good practice.

## Expected behavior
Wouldn't it be better if we could unit test a query with a mocked store instead of an actual store?

**Comment by NetanelBasal** [comment]
Yes, it's Ok. You are creating a new fresh store each time.
